classes/output/login.php

```diff
@@ -62,9 +62,24 @@ public static function output_login_options($wantsurl) {
                     $output .= \html_writer::link($idp['url'], $name, $attributes);
 
                     $output .= \html_writer::end_tag('div');
+                    $idploginpath = $idp['url'];
+                    $count++;
+                }
+            }
+        }
+        // Check if catadmin auth cookie set.
+        if (!NO_MOODLE_COOKIES) {
+            if (!empty($_COOKIE['catadmin_logout_cookie'])) {
+                if ($_COOKIE['catadmin_logout_cookie'] < (time() - DAYSECS)) {
+                    $catadmincookieset = 1;
```

Comment:
we shouldn't need this chunk in the code here - the way I'd expect this to work is that the catadmin auth would just return the button when calling authplugin->loginpage_idp_list() which would bump the number of buttons being displayed by one?

Reply:
I have updated it.